HopsanCore/src/ComponentSystem.cpp

```diff
@@ -1871,12 +1871,22 @@ bool ComponentSystem::checkModelBeforeSimulation()
                 break;
             }
 
-            // If a system parameter is used in the component, then erase it from the list
-            std::vector<HString>::iterator itp = std::find(unusedSysParNames.begin(), unusedSysParNames.end(), pCompParameters->at(p)->getValue());
-            if(itp != unusedSysParNames.end())
+            // If a system parameter is used in the component, then erase it from the list.
+            // To match only complete parameter names, prepend and append a space to the expression and replace operators by a space.
+            // Then find parameter names in expression.
+            HString expression = (" " + pCompParameters->at(p)->getValue() + " ")
+                    .replace("+", " ")
+                    .replace("-", " ")
+                    .replace("*", " ")
+                    .replace("/", " ")
+                    .replace("^", " ")
+                    .replace("(", " ")
+                    .replace(")", " ");
```

Comment:
Numhop (should be called embedded script) also supports < > | & operators,
see https://github.com/peterNordin/libNumHop/blob/master/README.md

Reply:
Yes, I need to update the documentation. And I agree, libnumhop should be able to extract variable names. I will add an issue to implement  that.